src/ptn.rs

```diff
@@ -105,6 +105,48 @@ impl FromStr for Square {
     }
 }
 
+impl Square {
+    const fn assert_on_board(&self, board_size: u8) {
+        assert!(self.row < board_size);
+        assert!(self.column < board_size);
+    }
+
+    /// Generate a Square that is one step in the direction from this Square.
+    /// If the generated Square is outside of the board, then this returns None instead.
+    pub fn checked_step(self, direction: Direction, board_size: u8) -> Option<Self> {
+        self.assert_on_board(board_size);
+        use Direction::*;
+        let (column, row) = (self.column, self.row);
+        let (column, row) = match direction {
+            Up => (column, row.checked_sub(1)?),
+            Down => (column, row.checked_add(1).filter(|&y| y < board_size)?),
```

Comment:
Up is +y, down is -y; also check rotations

Reply:
Good catch. This was actually a mistake.
Rotations are correct.